test/Gen.hs

```diff
@@ -130,15 +147,68 @@ w = genT
 
 type W = T "W"
 
+unit :: Gen ()
+unit = atom "()" ()
+
+identity :: Gen (a -> Identity a)
+identity = atom "Identity" Identity
+
 fn :: (Fn.Arg a, Fn.Vary a, Show a) => Gen b -> Gen (a -> b)
 fn b = Gen (lift (fmap (fmap runTerm) . showingFn <$> Fn.fn (fst <$> runWriterT (runGen b))))
 
+termFn :: Gen b -> Gen (a -> b)
+termFn b = Gen $ recursive Hedgehog.Gen.choice
+  [ runGen (atom "const" const <*> b) ]
+  []
+
 choice :: [Gen a] -> Gen a
 choice = Gen . Hedgehog.Gen.choice . Prelude.map runGen
 
 integral :: (Integral a, Show a) => Range a -> Gen a
 integral range = Gen (showing <$> Hedgehog.Gen.integral range)
 
+unicode :: Gen Char
+unicode = Gen (showing <$> Hedgehog.Gen.unicode)
+
+string :: Range Int -> Gen Char -> Gen String
+string range cs = Gen (showing <$> Hedgehog.Gen.string range (runTerm <$> runGen cs))
+
+
+newtype Run f g m = Run (forall a . f (m a) -> PureC (g a))
```

Comment:
This captures the shape of the handler function passed to the `Monad` & `MonadFix` tests.

Reply:
Can you move this comment (and your other explanatory comments about `RunL` / `RunR` / etc) into comments in the source file? They’re very helpful to have around.